### ISSUE TITLE
Switch statement doesn't think int.maxvalue is constant

### ISSUE THREAD

**Issue by Xyncgas** [dotnet/roslyn#58056]
### Description

![image](https://user-images.githubusercontent.com/35312807/144170235-9db46f7d-a5b0-4b87-85a6-6ca34a13a58f.png)

### Reproduction Steps

```
 BigInteger cycles = BigInteger.Pow(2, Dimension);
                    switch (cycles)
                    {
                        case < int.MaxValue:
                            //do something
                    }
```

### Expected behavior

case < int.MaxValue: shouldn't be an error when compiling

### Actual behavior

IDE highlights case < int.MaxValue:
compilation failed

**Comment by GrabYourPitchforks** [comment]
Reopening this issue for consideration by the Roslyn team. The error is correct, but the error text & VS's helpful screenshot (see first message in this thread) are counterintuitive. The error text leads to user to believe that `int.MaxValue` is not constant, which it clearly is. It's not immediately apparent that an implicit cast is taking place under the covers.

Is it worth considering updating the error text here to provide a better user experience?

**Comment by jcouv** [comment]
Right, the issue is that `int.MaxValue` is converted to `BigInteger` which results in a value that's not a constant. 
`cycles is (BigInteger)int.MaxValue` would result in the same error.
But the conversion is implicit and the error message doesn't call out the existence of the conversion...

[Repro](https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4FgBQABAzAAgGcAXAJwFcBjY/AYXywG8t9X9sAWfAWQAoBKFm2aY2YgPr4AvPmIALKIXyL8ABgDc+APRb8AU1KkA9qToBlVQEYArKpD4AgvkpGYJCDBoA3CABtyespKegAeAA561HoIQqwAvrHsBNiWAGzKYGG+UJRQNEYRpBDEJnS8sDRQ/NIAfLJyxgDu+DDkvr7qWHFAA):
```
public struct C 
{
    void M()
    {
       _ = this is 0; // error CS0150: A constant value is expected
    }
    public static implicit operator C(int i) => throw null;
}
```
